tortoisebeacon/tortoise_beacon.go

```diff
@@ -252,14 +235,23 @@ func (tb *TortoiseBeacon) Close() error {
 
 // GetBeacon returns a Tortoise Beacon value as []byte for a certain epoch or an error if it doesn't exist.
 func (tb *TortoiseBeacon) GetBeacon(epochID types.EpochID) ([]byte, error) {
+	if epochID == 0 {
+		return nil, ErrZeroEpoch
+	}
+
 	if tb.tortoiseBeaconDB != nil {
-		if val, ok := tb.tortoiseBeaconDB.GetTortoiseBeacon(epochID - 1); ok {
+		val, err := tb.tortoiseBeaconDB.GetTortoiseBeacon(epochID - 1)
+		if err == nil {
 			return val.Bytes(), nil
 		}
+
+		if !errors.Is(err, database.ErrNotFound) {
```

Comment:
if the error is not `database.ErrNotFound` we need to calculate it's value ?
If this is the expected behaviour add it as a comment

Reply:
I think that an error other than `database.ErrNotFound` should mean that something is wrong with the DB and should not normally happen. In this case, beacon still may be calculated. I'm not sure if we need to calculate it in this case though. I'll return an error in this case.